docs/development/developer.rst

```diff
@@ -4,69 +4,108 @@ Developer Guide
+
+    ``data.rollout.rollout``: Generates rollout by taking in any policy as input along with the environment. 
+
+.. Envs
+.. ----
+.. .. automodule:: imitation.envs
+..     :noindex:
+
+.. Policies
+.. --------
+.. .. automodule:: imitation.policies
+..     :noindex:
 
 Rewards
 -------
 .. automodule:: imitation.rewards
     :noindex:
 
+    ``rewards.reward_wrapper.RewardVecEnvWrapper``: This class wraps a ``VecEnv`` with a custom ``RewardFn``. 
+    The default reward function of the environment is overridden with the passed reward function 
+    and the original rewards are stored in the ``info_dict`` with the ``original_env_rew`` key. 
+    This class is used to override the original reward function of an environment with a learned 
+    reward function from the reward learning algorithms like preference comparisons.
 
 Scripts
 -------
-.. automodule:: imitation.scripts
-    :noindex:
+.. .. automodule:: imitation.scripts
+..     :noindex:
+
+We use Sacred to provide a command-line interface to run the experiments. The scripts to run the end-to-end experiments are
+available in ``scripts/``. You can take a look at the following doc links to understand how to use Sacred:
+
+- `Experiment Overview <https://sacred.readthedocs.io/en/stable/experiment.html>`_: Explains how to create and run experiments. All the experiment objects are defined in ``scripts/config`` in the respective algorithm name file and their ``main`` functions are defined in algorithm name file in ``scripts/``. For example, ``train_rl_ex`` object is defined in ``scripts.config.train_rl`` and its main function is in ``scripts.train_rl``.
+
+- `Ingredients <https://sacred.readthedocs.io/en/stable/ingredients.html>`_: Explains how to use ingredients to avoid code duplication across experiments. The ingredients used in our experiments are defined in ``scripts/common/``:
+  
+  .. autosummary:: 
+    imitation.scripts.common.common
+    imitation.scripts.common.demonstrations
+    imitation.scripts.common.reward
+    imitation.scripts.common.rl
+    imitation.scripts.common.train
+    imitation.scripts.common.wb    
+
+- `Configurations <https://sacred.readthedocs.io/en/stable/configuration.html>`_: Explains how to use configurations to parametrize runs. The configurations for different algorithms are defined in their file in ``scripts/``. Some of the commonly used configs and ingredients used across algorithms are defined in ``scripts/common/``.
+
+- `Command-Line Interface <https://sacred.readthedocs.io/en/stable/command_line.html>`_: Explains how to run the experiments through the command-line interface. Also note the section on how to `print configs <https://sacred.readthedocs.io/en/stable/command_line.html#print-config>`_ to verify the configurations used for the run.
+
+- `Controlling Randomness <https://sacred.readthedocs.io/en/stable/randomness.html>`_: Explains how to control randomness by seeding experiments through Sacred.
+
+.. Util
```

Comment:
I think it's worth having a `util` section. Needn't describe all of it, but `imitation.util.logger` is actually both quite important (used all over the code) and a common point of confusion for people. Could also give one-sentence descriptions of the other modules.